localstack/services/plugins.py

```diff
@@ -142,16 +162,361 @@ def is_enabled(self, api_names=None):
         return self.name() in api_names
 
 
-def register_plugin(plugin):
-    existing = SERVICE_PLUGINS.get(plugin.name())
-    if existing:
-        if existing.priority > plugin.priority:
-            return
-    SERVICE_PLUGINS[plugin.name()] = plugin
+class ServiceState(Enum):
+    UNKNOWN = "unknown"
+    AVAILABLE = "available"
+    DISABLED = "disabled"
+    STARTING = "starting"
+    RUNNING = "running"
+    # UPDATING = "updating"
+    STOPPING = "stopping"
+    STOPPED = "stopped"
+    ERROR = "error"
+
+
+class Journal:
+    name: str
+
+
+class ServiceContainer:
+    """
+    Holds a service, its state, and exposes lifecycle methods of the service.
+    """
+
+    service: Service
+    state: ServiceState
+    lock: threading.RLock
+    errors: List[Exception]
+
+    def __init__(self, service: Service, state=ServiceState.UNKNOWN):
+        self.service = service
+        self.state = state
+        self.lock = threading.RLock()
+        self.errors = list()
+
+    def get(self) -> Service:
+        return self.service
+
+    def start(self) -> bool:
+        try:
+            self.state = ServiceState.STARTING
+            self.service.start(asynchronous=True)
+            self.service.check(print_error=True)
+            self.state = ServiceState.RUNNING
+            return True
+        except Exception as e:
+            self.state = ServiceState.ERROR
+            self.errors.append(e)
+            LOG.error("error while starting service %s: %s", self.service.name(), e)
+            return False
+
+    def check(self) -> bool:
+        try:
+            self.service.check(print_error=True)
+            return True
+        except Exception:
+            return False
+
+    def stop(self):
+        try:
+            self.state = ServiceState.STOPPING
+            # TODO: actually stop the service
```

Comment:
Is the stop lifecycle phase being implemented in a later point in time? Otherwise this might need to be addressed :)